src/ts/background/index.ts

```diff
@@ -13,3 +16,106 @@ wrapStore(store, {
   // nd views such as browser tabs.
   portName: 'ExPort'
 })
+
+// listen to the port
+browser.runtime.onConnect.addListener(function (port) {
+  if (port.name !== '__SPECKLE__') return
+  port.onMessage.addListener(function (msg) {
+    switch (msg.method) {
+      case FUNCS.IS_LOCKED:
+        port.postMessage({ method: FUNCS.IS_LOCKED, result: keyringVault.isLocked() })
+        break
+      case FUNCS.IS_UNLOCKED:
+        port.postMessage({ method: FUNCS.IS_UNLOCKED, result: keyringVault.isUnlocked() })
+        break
+      case FUNCS.UNLOCK:
+        keyringVault.lock()
```

Comment:
does it need `await` before posting message?

Reply:
I don't think so. We just need to make sure the listener is added before sending the message to background, right?

Reply:
i meant when popup page send unlock message. do we want to wait until `keyringVault.lock()` is completed then post message back. we may also want to try/catch here

Reply:
don't worry about it. lock is not an async

Reply:
Fixed typo, it should FUNCS.LOCK here